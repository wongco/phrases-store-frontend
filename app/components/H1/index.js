import styled from 'styled-components';

const H1 = styled.h1`
  text-align: center;
  font-size: 200%;
  font-weight: 200;
`;

export default H1;
