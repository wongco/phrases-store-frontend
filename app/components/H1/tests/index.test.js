import React from 'react';
import { shallow, mount } from 'enzyme';
import { enzymeFind } from 'styled-components/test-utils';
import toJson from 'enzyme-to-json';
import H1 from '../index';

describe('<H1 />', () => {
  it('renders without crashing', () => {
    shallow(<H1 />);
  });

  it('matches snapshot for default criteria', () => {
    const wrapper = shallow(<H1 />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('should render a prop', () => {
    const id = 'testId';
    const wrapper = mount(<H1 id={id} />);
    const renderedComponent = enzymeFind(wrapper, H1);
    expect(renderedComponent.prop('id')).toBe('testId');
  });

  it('should render its text', () => {
    const children = 'Text';
    const wrapper = mount(<H1>{children}</H1>);
    const renderedComponent = enzymeFind(wrapper, H1);
    expect(renderedComponent.contains(children)).toBe(true);
  });
});
