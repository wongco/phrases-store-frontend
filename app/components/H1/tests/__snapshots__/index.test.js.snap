// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H1 /> matches snapshot for default criteria 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "H1-tncdg6-0",
        "isStatic": true,
        "rules": Array [
          "text-align:center;font-size:200%;font-weight:200;",
        ],
      },
      "displayName": "H1",
      "render": [Function],
      "styledComponentId": "H1-tncdg6-0",
      "target": "h1",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
